src/fields/Fields.cpp

```diff
@@ -507,30 +507,28 @@ Fields::Copy (const int lev, const int i_slice, const amrex::Geometry& diag_geom
 }
 
 void
-Fields::ShiftSlices (int nlev, int islice, amrex::Geometry geom, amrex::Real patch_lo,
+Fields::ShiftSlices (int lev, int islice, amrex::Geometry geom, amrex::Real patch_lo,
                      amrex::Real patch_hi)
 {
     HIPACE_PROFILE("Fields::ShiftSlices()");
 
-    for (int lev = 0; lev < nlev; ++lev) {
-
     // skip all slices which are not existing on level 1
     if (lev == 1) {
         // use geometry of coarse grid to determine whether slice is to be solved
         const amrex::Real* problo = geom.ProbLo();
         const amrex::Real* dx = geom.CellSize();
         const amrex::Real pos = (islice+0.5_rt)*dx[2]+problo[2];
-        if (pos < patch_lo || pos > patch_hi) continue;
+        if (pos < patch_lo || pos > patch_hi) return;
     }
 
     const bool explicit_solve = Hipace::GetInstance().m_explicit;
-    const bool mesh_refinement = nlev != 1;
+    const bool mesh_refinement = Hipace::GetInstance().maxLevel() != 0;
```

Comment:
Should we define such a variable, if used in other places? Just a suggestion, ofc this is not relevant if that's the only place.
```suggestion
    const bool mesh_refinement = Hipace::m_do_MR;
```

Reply:
This would indeed be useful. Should I do that in this PR?